stablehlo/reference/Ops.cpp

```diff
@@ -68,6 +81,43 @@ Tensor eval(FloorOp op, const Tensor &operand) {
   return result;
 }
 
+Tensor eval(IotaOp op) {
+  Tensor result(op.getType());
+  Type elType = result.getType().getElementType();
+  uint64_t iotaDimension = op.getIotaDimension();
+  for (auto it = result.index_begin(); it != result.index_end(); ++it) {
+    auto iota = (*it)[iotaDimension];
+    if (isSupportedSignedIntegerType(elType)) {
+      result.set(*it, Element(elType, APInt(elType.getIntOrFloatBitWidth(),
+                                            iota, /*isSigned=*/true)));
+    } else if (isSupportedUnsignedIntegerType(elType)) {
+      result.set(*it, Element(elType, APInt(elType.getIntOrFloatBitWidth(),
+                                            iota, /*isSigned=*/false)));
+    } else if (isSupportedFloatType(elType)) {
+      APFloat val = APFloat((double)iota);
+      bool roundingErr;
+      val.convert(elType.dyn_cast<FloatType>().getFloatSemantics(),
```

Comment:
`.cast<FloatType>` to express the fact that we know that this cast will succeed.